src/Network/NWEndpoint.cs

```diff
@@ -183,5 +184,76 @@ public class NWEndpoint : NativeObject {
 		[iOS (13,0)]
 #endif
 		public string? Url => Marshal.PtrToStringAnsi (nw_endpoint_get_url (GetCheckedHandle ()));
+
+
+#if NET
+		[SupportedOSPlatform ("tvos16.0")]
+		[SupportedOSPlatform ("macos13.0")]
+		[SupportedOSPlatform ("ios16.0")]
+		[SupportedOSPlatform ("maccatalyst16.0")]
+#else
+		[TV (16,0)]
+		[Mac (13,0)]
+		[iOS (16,0)]
+		[Watch (9,0)]
+#endif
+		[DllImport (Constants.NetworkLibrary)]
+		static extern unsafe byte* nw_endpoint_get_signature (OS_nw_endpoint endpoint, out nuint out_signature_length);
+
+#if NET
+		[SupportedOSPlatform ("tvos16.0")]
+		[SupportedOSPlatform ("macos13.0")]
+		[SupportedOSPlatform ("ios16.0")]
+		[SupportedOSPlatform ("maccatalyst16.0")]
+#else
+		[TV (16,0)]
+		[Mac (13,0)]
+		[iOS (16,0)]
+		[Watch (9,0)]
+#endif
+		public ReadOnlySpan<byte> Signature {
+			get {
+				unsafe {
+					var data = nw_endpoint_get_signature (GetCheckedHandle (), out var length);
```

Comment:
What's the lifetime of the returned pointer? Do we have to free it?

Reply:
Headers are not very clear:
```
Returns a pointer to a buffer of bytes containing a signature, or NULL if the endpoint is not signed. 
```

Reply:
I think we should take the safe route then, and copy the bytes instead (and not free them).